lib/rubygems/commands/query_command.rb

```diff
@@ -86,7 +86,7 @@ def execute
       name = Array(options[:name])
     else
       args = options[:args].to_a
-      name = options[:exact] ? args : args.map{|arg| /#{arg}/i }
+      name = options[:exact] ? args.map{|arg| /\A#{arg}\Z/ } : args.map{|arg| /#{arg}/i }
```

Comment:
Yeah, that makes sense to me.  This way it's treated like a pure string, whereas the ternary else condition there allows a user to pass an actual regex.